src/wallet/wallettool.cpp

```diff
@@ -106,6 +107,12 @@ bool ExecuteWalletToolFunc(const std::string& command, const std::string& name)
 {
     fs::path path = fs::absolute(name, GetWalletDir());
 
+    // -dumpfile is only allowed with dump and createfromdump. Disallow it for all other commands.
+    if (gArgs.IsArgSet("-dumpfile") && command != "dump" && command != "createfromdump") {
```

Comment:
In commit "wallettool: Add dump command" (e1e7a90d5f0616a46ffadd62a9f1c65406cca6b4)

For a followup, it would be good to print an error in the `if (gArgs.IsArgSet("-descriptors") && command != "create")` case so there is an explicit error if someone tries to pass `-descriptors`/`-nodescriptors` to `createfromdump`. (Suggested originally in S3RK's PR https://github.com/bitcoin/bitcoin/pull/20365#discussion_r534395176)